src/components/file/file.service.ts

```diff
@@ -210,8 +210,10 @@ export class FileService {
     session: ISession
   ): Promise<File> {
     let upload;
+    let rootUpload;
     try {
       upload = await this.bucket.headObject(`temp/${uploadId}`);
+      rootUpload = await this.bucket.headObject(`${uploadId}`);
```

Comment:
Ok so this is the tricky part. Both of these calls throw exceptions but we want to catch and handle both of those together. i.e. both of the calls do _not_ have to be successful - one should fail most of the time even.

We can use `Promise.allSettled` to facilitate this. This also calls both in parallel which is a nice bonus.
```tsx
const [tempUpload, existingUpload] = await Promise.allSettled([
  this.bucket.headObject(`temp/${uploadId}`),
  this.bucket.headObject(uploadId),
]);
```
This replaces the try/catch block btw.

Both of the resulting items have a `status` of either `'fulfilled'` or `'rejected'` and either a `value` or `reason` based on the status. TS has this typed.

From there the logic you added below can be adapted to this.

Reply:
@CarsonF Thanks!!